modules/exploits/multi/http/solr_velocity_rce.rb

```diff
@@ -0,0 +1,377 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/exploit/powershell'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Apache Solr Remote Code Execution via Velocity Template',
+        'Description'    => %q(
+          This module exploits a vulnerability in Apache Solr <= 8.2.0 which allows remote code execution via a custom
+          Velocity template.
+
+          From the Tenable advisory:
+          An attacker could target a vulnerable Apache Solr instance by first identifying a list
+          of Solr core names. Once the core names have been identified, an attacker can send a specially crafted
+          HTTP POST request to the Config API to toggle the params resource loader value for the Velocity Response
+          Writer in the solrconfig.xml file to true. Enabling this parameter would allow an attacker to use the Velocity
+          template parameter in a specially crafted Solr request, leading to RCE. This vulnerability currently does not
+          have a CVE number assigned.
+
+          Payload defaults to x86 Meterpreter reverse TCP for Windows and plain old reverse TCP shell for Linux.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            's00py', # Discovery and PoC
+            'jas502n', # exploit code on Github
+            'AleWong', # ExploitDB contribution, and exploit code on Github
+            'Imran E. Dawoodjee <imran[at]threathounds.com>' # Metasploit module
+          ],
+        'References'     =>
+            [
+              [ 'EDB', '47572' ],
+              [ 'URL', 'https://www.tenable.com/blog/apache-solr-vulnerable-to-remote-code-execution-zero-day-vulnerability'],
+              [ 'URL', 'https://www.huaweicloud.com/en-us/notice/2018/20191104170849387.html'],
+              [ 'URL', 'https://gist.github.com/s00py/a1ba36a3689fa13759ff910e179fc133/'],
+              [ 'URL', 'https://github.com/jas502n/solr_rce'],
+              [ 'URL', 'https://github.com/AleWong/Apache-Solr-RCE-via-Velocity-template'],
+            #[ 'CVE', 'CVE-1111-2222'],
```

Comment:
```suggestion
            #[ 'CVE', 'CVE-1111-2222'],
```

I'm guessing a CVE won't be issued for this one.

Reply:
Fixed in commit ce011375252e6b61502c7f33e58024e7fcbbf906.